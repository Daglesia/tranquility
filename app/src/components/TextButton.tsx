export interface TextButtonProps {
  text: string;
}

export default function TextButton({ text }: TextButtonProps) {
  return (
    <button className="relative min-w-48 w-fit h-14 bg-secondary rounded-3xl flex place-items-center justify-center cursor-pointer hover:bg-secondary__variant--dark">
      <p className="text-h3 font-extrabold text-background px-8">{text}</p>
    </button>
  );
}
