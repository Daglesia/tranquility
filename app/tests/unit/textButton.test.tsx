import { expect, it, describe, beforeAll } from "vitest";
import { render, RenderResult, screen } from "@testing-library/react";
import TextButton, { TextButtonProps } from "@/components/TextButton";

const BUTTON_PROPS: TextButtonProps = {
  text: "textbutton",
};

describe("TextButton", () => {
  let button: RenderResult;
  beforeAll(() => {
    button = render(<TextButton {...BUTTON_PROPS} />);
  });

  it("matches snapshot", () => {
    expect(button).toMatchSnapshot();
  });

  it("renders text", () => {
    expect(screen.getByRole("paragraph").innerHTML).toContain(
      BUTTON_PROPS.text,
    );
  });
});
