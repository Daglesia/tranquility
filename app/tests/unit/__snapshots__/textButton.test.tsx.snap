// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TextButton > matches snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="relative min-w-48 w-fit h-14 bg-secondary rounded-3xl flex place-items-center justify-center cursor-pointer hover:bg-secondary__variant--dark"
      >
        <p
          class="text-h3 font-extrabold text-background px-8"
        >
          textbutton
        </p>
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="relative min-w-48 w-fit h-14 bg-secondary rounded-3xl flex place-items-center justify-center cursor-pointer hover:bg-secondary__variant--dark"
    >
      <p
        class="text-h3 font-extrabold text-background px-8"
      >
        textbutton
      </p>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
